hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneFileSystem.java

```diff
@@ -773,7 +787,7 @@ numEntries, uri, workingDir, getUsername())
       // exhausted.
     } while (tmpStatusList.size() == numEntries);
 
-    return statuses.toArray(new FileStatus[0]);
+    return statuses.toArray(new FileStatusAdapter[0]);
```

Comment:
is it possible to return statuses instead? Save one temporary array allocation.

Reply:
ohh!! yes!!